composer update with version 1 throws AssertionError

### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 2.0.0beta1

When using `2.0.0beta1` and issuing a `composer update` with version 1 of composer, it throws an `AssertionError`:

`PHP Fatal error:  Uncaught AssertionError: assert($rewriter instanceof DependencySolvingCapableInterface) in /mnt/e/PhpStormProjects/laminas/test-mvc-co2/vendor/laminas/laminas-dependency-plugin/src/DependencyRewriterPluginDelegator.php:69`

It only occurs if the corresponding `assert.` settings in `php.ini` are active.

The execution itself works because `DependencyRewriterV1` _does_ implement the method `onPreDependenciesSolving` which `DependencySolvingCapableInterface` requires.

It seems, that adding `implements DependencySolvingCapableInterface` to `DependencyRewriterV1` solves the error.

--- a/src/DependencyRewriterV1.php
+++ b/src/DependencyRewriterV1.php
@@ -18,7 +18,7 @@ use function get_class;
 use function in_array;
 use function sprintf;
 
-final class DependencyRewriterV1 extends AbstractDependencyRewriter
+final class DependencyRewriterV1 extends AbstractDependencyRewriter implements DependencySolvingCapableInterface
 {
     /**
      * Replace ZF packages present in the composer.json during install or

--- a/test/DependencyRewriterV1Test.php
+++ b/test/DependencyRewriterV1Test.php
@@ -21,6 +21,7 @@ use Composer\Plugin\PluginInterface;
 use Composer\Plugin\PreCommandRunEvent;
 use Composer\Repository\RepositoryManager;
 use Laminas\DependencyPlugin\DependencyRewriterV1;
+use Laminas\DependencyPlugin\DependencySolvingCapableInterface;
 use PHPUnit\Framework\TestCase;
 use Prophecy\Argument;
 use Prophecy\PhpUnit\ProphecyTrait;
@@ -739,5 +740,10 @@ final class DependencyRewriterV1Test extends TestCase
 
         $this->assertNull($this->plugin->onPrePackageInstallOrUpdate($event->reveal()));
         $this->assertSame($replacementPackage, $operation->getTargetPackage());
+    }
+
+    public function testRewriterImplementsDependencySolvingCapableInterface() : void
+    {
+        self::assertInstanceOf(DependencySolvingCapableInterface::class, $this->plugin);
     }
 }

